src/reducers/web3.js

```diff
@@ -30,22 +30,41 @@ export const readSmartContract = address => async (dispatch, getState) => {
   const transparencyDecision = computeICOTransparency(answers)[0];
 
   dispatch(setProperties(address, { decision: transparencyDecision }));
-  const parameters = await getICOParameters(web3, address);
+  const icoParameters = await getICOParameters(web3, address);
+  const parameters = icoParameters[0];
+  const smartContractDictionary = icoParameters[1];
+
   config.ICOs[address].decimals = parameters.decimals || config.ICOs[address].decimals; // set decimals in config from smart contract
   Object.keys(parameters).forEach((par) => {
     const parameter = parameters[par];
     if (parameter === null) return;
     const tempResult = {};
-    if (typeof parameter === 'object' && typeof parameter.then === 'function') {
+    if (smartContractDictionary[par] === 'bytes32') {
+      const asciiValue = web3.toAscii(parameter);
+        // check if it has value
+      const characters = [];
```

Comment:
use regex to trim trailing 0s as here https://stackoverflow.com/questions/26156292/trim-specific-character-from-a-string